### PR TITLE
Snyder_l221_land_input_1

### PR DESCRIPTION
**Pull request by abigailsnyder**
#564

### CODE REVIEW

**Comment by bpbond** [comment]
Can also tag a non-aglu person if you are particularly busy or would rather now @rynacui .

File: R/constants.R [hunk 1]
```diff
@@ -179,6 +179,13 @@ aglu.WOOD_ENERGY_CONTENT_GJKG <- 0.0189
 # wood water content
 aglu.WOOD_WATER_CONTENT <- 0.065
 
+# Min veg and soil carbon densities
```
Comment (bpbond) [inline R/constants.R]: Units for these?

**Comment by rynacui** [comment]
No worries! Finishing up my chunk and I can help here.

File: R/constants.R [hunk 2]
```diff
@@ -170,7 +170,7 @@ aglu.FOREST_HARVEST_INDEX <- 0.8
 # Forest Erosion Control in kg/m^2
 aglu.FOREST_EROSION_CTRL_KGM2 <- 0.2
 
-#Mill Erosion Control in kg/m^2
+# Mill Erosion Control in kg/m^2
```
Comment (bpbond) [inline R/constants.R]: God there's some weird stuff in gcam.

File: R/module-helpers.R [hunk 3]
```diff
@@ -278,6 +278,55 @@ replace_GLU <- function(d, map, GLU_pattern = "^GLU[0-9]{3}$") {
...
+  if (!("region" %in% names(carbon_info_table))){
+    carbon_info_table %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") ->
+      carbon_info_table
+  }
+
+  # data[ c( "hist.veg.carbon.density", "hist.soil.carbon.density", "mature.age" ) ] <-
+  #   carbon_info_table[ match( vecpaste( data[ data_matchvars ] ),
+  #                             vecpaste( carbon_info_table[ carbon_info_matchvars ] ) ),
+  #                      carbon_info_table_names ]
```
Comment (bpbond) [inline R/module-helpers.R]: Remove

File: R/zchunk_L221.land_input_1.R [hunk 4]
```diff
@@ -1,20 +1,38 @@
 #' module_aglu_L221.land_input_1
 #'
-#' Briefly describe what this chunk does.
+#' This chunk produces L221.LN0_Logit - Logit exponent of the top-level (zero) land nest by region; L221.LN0_Land - Total regional land allocation in top level nest;
+#' L221.LN0_SoilTimeScale - Soil time scale by region; L221.LN1_ValueLogit - Unmanaged land value by region and GLU, and logit exponent of first nest;
+#' L221.LN1_HistUnmgdAllocation - Historical land cover fora unmanaged land (LT_GLU) in the first nest by region;
+#' L221.LN1_UnmgdAllocation - Land cover in the model base periods for unmanaged land (LT_GLU) in the first nest by region; and
+#' L221.LN1_UnmgdCarbon - Carbon content for unmanaged land (LT_GLU) in first nest by region.
```
Comment (bpbond) [inline R/zchunk_L221.land_input_1.R]: I'd prefer the top-level description be a single sentence or two. Save this for the `@details` please.

File: R/zchunk_L221.land_input_1.R [hunk 5]
```diff
@@ -1,20 +1,38 @@
...
 #' @param command API command to execute
 #' @param ... other optional parameters, depending on command
 #' @return Depends on \code{command}: either a vector of required inputs,
 #' a vector of output names, or (if \code{command} is "MAKE") all
 #' the generated outputs: \code{curr_table$data}, \code{L221.LN0_Logit}, \code{L221.LN0_Land}, \code{L221.LN0_SoilTimeScale}, \code{curr_table$data}, \code{L221.LN1_ValueLogit}, \code{L221.LN1_HistUnmgdAllocation}, \code{L221.LN1_UnmgdAllocation}, \code{L221.LN1_UnmgdCarbon}. The corresponding file in the
 #' original data system was \code{L221.land_input_1.R} (aglu level2).
-#' @details Describe in detail what this chunk does.
+#' @details  L221.LN0_Logit Logit exponent of the top-level (zero) land nest by region.
+#' AgLu regions are given externally defined constant logit information.
+#' L221.LN0_Land Total regional land allocation in top level nest.
```
Comment (bpbond) [inline R/zchunk_L221.land_input_1.R]: May want to use an [itemized list](https://stackoverflow.com/questions/9267584/when-documenting-in-roxygen-how-do-i-make-an-itemized-list-in-details) here.

File: R/zchunk_L221.land_input_1.R [hunk 6]
```diff
@@ -28,11 +46,10 @@ module_aglu_L221.land_input_1_DISABLED <- function(command, ...) {
              "L125.LC_bm2_R",
              "L131.LV_USD75_m2_R_GLU"))
   } else if(command == driver.DECLARE_OUTPUTS) {
-    return(c("curr_table$data",
+    return(c(#"curr_table$data",
```
Comment (bpbond) [inline R/zchunk_L221.land_input_1.R]: Delete

File: R/zchunk_L221.land_input_1.R [hunk 7]
```diff
@@ -54,124 +71,222 @@ module_aglu_L221.land_input_1_DISABLED <- function(command, ...) {
...
+    # Build L221.LN1_ValueLogit: Unmanaged land value by region and AEZ, and logit exponent of first nest
+    L125.LC_bm2_R_LT_Yh_GLU %>%
+      left_join(select(A_LT_Mapping, Land_Type, LandNode1), by = "Land_Type") %>%
+      select(region, GLU, LandNode1) %>%
+      distinct %>%
+      na.omit %>%
+      mutate(LandAllocatorRoot = "root",
+             logit.year.fillout = min(BASE_YEARS))  %>%
+      # add exponents and logit types
+      left_join(A_LandNode_logit, by = c("LandNode1" = "LandNode")) %>%
```
Comment (bpbond) [inline R/zchunk_L221.land_input_1.R]: Here and below, I assume there's a reason we're not using `left_join_error_no_match`?

**Comment by abigailsnyder** [comment]
I'm fixing the test that is causing me to fail on travis; looks like the Old DS file just has a bunch of rows of all NAs. So I'm getting the "wrong" number of rows when the new DS compares.

File: R/constants.R (hunk 1 above)
Comment (abigailsnyder) [inline R/constants.R]: no clue @pkyle ?

Comment (pkyle) [inline R/constants.R]: Unitless (mass of water / total wood mass)

Comment (abigailsnyder) [inline R/constants.R]: @pkyle any memory of aglu.MIN_VEG_CARBON_DENSITY and aglu.MIN_SOIL_CARBON_DENSITY?

I'm guessing (some unit of Carbon mass)/(some land area unit or some mass unit or some volume unit)

Comment (pkyle) [inline R/constants.R]: kg C per m2
Those are zero tho, right?

Comment (abigailsnyder) [inline R/constants.R]: yep, still helpful for comments I guess

**Comment by abigailsnyder** [comment]
looks like @bpbond comments addressed and travis is on the way to passing

File: R/zchunk_L221.land_input_1.R [hunk 8]
```diff
@@ -54,124 +70,223 @@ module_aglu_L221.land_input_1_DISABLED <- function(command, ...) {
...
+
+    # Carbon contents and mature ages
+    # Carbon content info, unmanaged land in the first nest
+    L221.LC_bm2_R_Unmgd1_Yh_GLU %>%
+      filter(year == max(BASE_YEARS)) %>%
+      select(-year, -allocation) %>%
+      left_join_error_no_match(GCAMLandLeaf_CdensityLT, by = c("Land_Type" = "LandLeaf")) %>%
+      rename(Cdensity_LT = Land_Type.y) %>%
+      add_carbon_info(., carbon_info_table = L121.CarbonContent_kgm2_R_LT_GLU) %>%
+      select(region, LandAllocatorRoot, LandNode1, UnmanagedLandLeaf, hist.veg.carbon.density, hist.soil.carbon.density, veg.carbon.density,
```
Comment (rynacui) [inline R/zchunk_L221.land_input_1.R]: maybe there is the `LEVEL2_DATA_NAMES` can be used?

File: R/zchunk_L221.land_input_1.R [hunk 9]
```diff
@@ -54,124 +70,223 @@ module_aglu_L221.land_input_1_DISABLED <- function(command, ...) {
     L125.LC_bm2_R <- get_data(all_data, "L125.LC_bm2_R")
     L131.LV_USD75_m2_R_GLU <- get_data(all_data, "L131.LV_USD75_m2_R_GLU")
 
-    # ===================================================
```
Comment (rynacui) [inline R/zchunk_L221.land_input_1.R]: silence package?

File: R/zchunk_L221.land_input_1.R (hunk 8 above)
Comment (bpbond) [inline R/zchunk_L221.land_input_1.R]: Good catch.

**Comment by abigailsnyder** [comment]
@bpbond - working on these final edits now. Would it make sense for someone (plz not me) to create a Level 2 Chunk wiki page? 
Detailing the parts to avoid (get_logit_function_table) a little more clearly, demonstrating LEVEL2_DATA_NAMES, etc?

**Comment by bpbond** [comment]
Thanks. Yes, agreed. I'll ask Pralit if he can do this.

**Comment by abigailsnyder** [comment]
I think all comments addressed @bpbond

**Pull request merged**